bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniDrugOrderServiceImpl.java

```diff
@@ -167,6 +178,110 @@ public List<Order> getAllDrugOrders(String patientUuid, String patientProgramUui
         return orderDao.getAllOrders(patientByUuid, orderTypeByUuid, conceptsForDrugs, drugConceptsToBeExcluded, encounters);
     }
 
+    @Override
+    public List<BahmniDrugOrder> getSortedBahmniDrugOrdersForVisit(String patientUuid, String visitUuid) {
+        List<DrugOrder> drugOrderList = getPrescribedDrugOrders(Arrays.asList(visitUuid), patientUuid, null,null, null, null, null);
+        List<BahmniDrugOrder> bahmniDrugOrderList = getBahmniDrugOrdersForVisit(patientUuid, drugOrderList);
+        Collections.sort(bahmniDrugOrderList, new Comparator<BahmniDrugOrder>() {
+            @Override
+            public int compare(BahmniDrugOrder o1, BahmniDrugOrder o2) {
+                return o1.getEffectiveStartDate().compareTo(o2.getEffectiveStartDate());
+            }
+        });
+        return bahmniDrugOrderList;
+    }
+
+    @Override
+    public Map<BahmniDrugOrder, Integer> getMergedDrugOrderMap(List<BahmniDrugOrder> drugOrderList) {
```

Comment:
is the duration always in days?

Reply:
refactored the logic to work for mixed duration unit types